corehq/form_processor/models/cases.py

```diff
@@ -294,6 +295,29 @@ def hard_delete_cases(self, domain, case_ids, *, publish_changes=True):
 
         return deleted_count
 
+    def hard_delete_cases_before_cutoff(self, cutoff, dry_run=True):
+        """
+        Permanently deletes cases with deleted_on set to a datetime earlier than
+        the specified cutoff datetime and creates a tombstone record of the deletion.
+        :param cutoff: datetime used to obtain the cases to be hard deleted
+        :param dry_run: if True, no changes will be committed to the database
+        and this method is effectively read-only
+        :return: dictionary of count of deleted objects per table
+        """
+        counts = {}
+        class_path = 'form_processor.CommCareCase'
+        for db_name in get_db_aliases_for_partitioned_query():
+            queryset = self.using(db_name).filter(deleted_on__lt=cutoff)
+            if dry_run:
+                deleted_counts = {class_path: queryset.count()}
+                counts.update(deleted_counts)
+            else:
+                for case in queryset:
+                    create_deleted_sql_doc(case.case_id, class_path, case.domain, case.deleted_on)
```

Comment:
Would it be possible to do this using `DeletedSQLDoc.objects.bulk_create()`? That would be much more efficient than creating them one at a time.